packages/eslint-plugin/docs/rules/no-unnecessary-type-parameters.md

```diff
@@ -0,0 +1,104 @@
+---
+description: 'Disallow type parameters that only appear once'
+---
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/no-unnecessary-type-parameters** for documentation.
+
+This rule forbids type parameters that only appear once in a function, method, or class definition.
+
+Type parameters relate two types. If a generic type parameter only appears once, then it is not relating anything. It can usually be replaced with explicit types such as `unknown`.
+
+At best unnecessary type parameters make code harder to read. At worst they can be used to disguise unsafe type assertions.
+
+## Options
+
+None.
+
+## Examples
+
+<!--tabs-->
+
+### ❌ Incorrect
+
+```ts
+function third<A, B, C>(a: A, b: B, c: C): C {
+  return c;
+}
+
+function parseYAML<T>(input: string): T {
+  // ...
+}
+
+function printProperty<T, K extends keyof T>(obj: T, key: K) {
+  console.log(obj[key]);
+}
+
+class Joiner<T extends string | number> {
+  join(els: T[]) {
+    return els.map(el => '' + el).join(',');
+  }
+}
+```
+
+### ✅ Correct
+
+```ts
+function third<A, B, C>(a: unknown, b: unknown, c: C): C {
+  return c;
+}
+
+function parseYAML(input: string): unknown {
+  // ...
+}
+
+function printProperty<T>(obj: T, key: keyof T) {
+  console.log(obj[key]);
+}
+
+class Joiner {
+  join(els: (string | number)[]) {
+    return els.map(el => '' + el).join(',');
+  }
+}
+
+// T appears twice: in the type of arg and as the return type
```

Comment:
[Praise] These two added valid examples are very clear & helpful for explaining the rule IMO. 🔥